lib/Driver/ToolChains.cpp

```diff
@@ -217,6 +217,7 @@ void ToolChain::addCommonFrontendArgs(const OutputInfo &OI,
   inputArgs.AddLastArg(arguments, options::OPT_profile_use);
   inputArgs.AddLastArg(arguments, options::OPT_profile_coverage_mapping);
   inputArgs.AddLastArg(arguments, options::OPT_warnings_as_errors);
+  inputArgs.AddLastArg(arguments, options::OPT_no_warnings_as_errors);
```

Comment:
I think you need to use `AddAllArgs` here, passing both `OPT_warnings_as_errors` and `OPT_no_warnings_as_errors`, so we don't perturb the order.